frontend/src/Pages/HomePage.js

```diff
@@ -20,19 +41,109 @@ export default function HomePage() {
     };
 
+    const validateNickname = (nickname) => {
+        let validCheck = true;
+        setNickname(nickname);
+        if (nickname === '') {
+            validCheck = false;
+            setnonEmpty(invalidNickname);
+            setTickNonempty(false);
+        } else {
+            setnonEmpty(validNickname);
+            setTickNonempty(true);
+        }
+        if (nickname.includes(' ')) {
+            validCheck = false;
+            setnoSpaces(invalidNickname);
+            setTickNoSpaces(false);
+        } else {
+            setnoSpaces(validNickname);
+            setTickNoSpaces(true);
+        }
+        if (nickname.length > 10) {
+            validCheck = false;
+            setLessThanTen(invalidNickname);
+            setTickLessThanTen(false);
+        } else {
+            setLessThanTen(validNickname);
+            setTickLessThanTen(true);
+        }
+        if (validCheck) {
+            setjoinDisabled(false);
+        } else {
+            setjoinDisabled(true);
+        }
+    };
+
     return (
         <div>
             <div className={styles.container}>
                 <h1>MAFIA</h1>
-                <TextField
-                    color="secondary"
-                    id="nickname"
-                    label="Enter Nickname"
-                    autoComplete="off"
-                    variant="outlined"
-                    value={nickname}
-                    onChange={(e) => setNickname(e.target.value)}
-                ></TextField>
+                <ClickAwayListener onClickAway={handleTooltipClose}>
+                    <div>
+                        <Tooltip
+                            PopperProps={{
+                                disablePortal: true,
+                            }}
+                            onClose={handleTooltipClose}
+                            open={open}
+                            disableFocusListener
+                            disableHoverListener
+                            disableTouchListener
+                            title={
+                                <React.Fragment>
+                                    <div className={styles.checkers}>
+                                        <li className={nonEmpty}>
+                                            {tickNonempty ? (
+                                                <span>
+                                                    <CheckRoundedIcon fontSize="small" />
+                                                    <span className={styles.checkitems}> Non-empty nickname</span>{' '}
+                                                </span>
+                                            ) : (
+                                                <span className={styles.checkitems}> Non-empty nickname</span>
+                                            )}
+                                        </li>
+                                        <li className={noSpaces}>
```

Comment:
Instead of creating a state variable just to hold the className you could use the `tickNoSpaces` to conditionally add classnames. You can do this via
`tickNoSpaces ?  styles.validNickname : styles.invalidNickname`
OR
use classnames import
`classNames({styles.validNickname: tickNoSpaces, styles.invalidNickname: !tickNoSpaces })`

Reply:
great suggestion